src/blob/index/core.rs

```diff
@@ -332,7 +332,17 @@ where
                 let records_allocated;
                 if let Some(v) = data.headers.get_mut(key) {
                     let old_capacity = v.capacity();
-                    v.push(h);
+                    // Keep ordered by timestamp
+                    let mut pos = 0;
+                    if v.len() > 4 {
```

Comment:
Does it provide any noticeable effects in benchmarks on pearl?

Reply:
In pearl benchmarks, this change is not noticable. However, from my experience: in micro-benchmarks linear search in arrays of small length is faster than binary search due to better cache usage